File: src/test/java/com/example/e2e/api/JsonPlaceHolderIT.java
package com.example.e2e.api;

import com.example.screenplay.action.GetAListOfAllPosts;
import com.example.screenplay.action.UploadNewPost;
import com.example.screenplay.question.api.NumberOfReturnedPosts;
import net.serenitybdd.junit5.SerenityJUnit5Extension;
import net.serenitybdd.screenplay.Actor;
import net.serenitybdd.screenplay.rest.abilities.CallAnApi;
import net.thucydides.core.annotations.Narrative;
import org.apache.http.HttpStatus;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import static net.serenitybdd.screenplay.GivenWhenThen.seeThat;
import static net.serenitybdd.screenplay.rest.questions.ResponseConsequence.seeThatResponse;
import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.Matchers.greaterThan;

@ExtendWith(SerenityJUnit5Extension.class)
@Narrative(title = "title for json placeholder tests", text = "FIXME description")
public class JsonPlaceHolderIT {

  private static Actor author;

  @BeforeAll
  public static void setUp() {
    author = new Actor("author");
    author.can(CallAnApi.at("https://jsonplaceholder.typicode.com"));
  }

  @Test
  public void whenThereAreManyPosts() {
    author.attemptsTo(new GetAListOfAllPosts());
    author.should(seeThat(new NumberOfReturnedPosts(), greaterThan(99)));
  }

  @Test
  public void whenPosting() {
    author.attemptsTo(
        UploadNewPost.containing("{\"title\":\"foo\",\"body\":\"bar\",\"userId\":1}"));

    author.should(seeThatResponse(response -> response.statusCode(HttpStatus.SC_CREATED)));
    author.should(seeThatResponse(response -> response.body("id", is(101))));
  }
}

File: src/test/kotlin/com/example/e2e/api/JsonPlaceHolderIT.kt
package com.example.e2e.api

import com.example.screenplay.action.GetListOfAllPosts
import com.example.screenplay.action.UploadNewPost
import com.example.screenplay.question.NumberOfReturnedPosts
import net.serenitybdd.junit.runners.SerenityRunner
import net.serenitybdd.screenplay.Actor
import net.serenitybdd.screenplay.GivenWhenThen.*
import net.serenitybdd.screenplay.rest.abilities.CallAnApi
import net.serenitybdd.screenplay.rest.questions.ResponseConsequence.*
import net.thucydides.core.annotations.Narrative
import org.apache.http.HttpStatus
import org.hamcrest.CoreMatchers
import org.hamcrest.Matchers.*
import org.junit.BeforeClass
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(SerenityRunner::class)
@Narrative(text = ["a simple REST API example"])
open class JsonPlaceHolderIT {

  @Test
  fun `when there are many posts`() {
    author.attemptsTo(GetListOfAllPosts())
    author.should(seeThat(NumberOfReturnedPosts(), greaterThan(99)))
  }

  @Test
  fun `when posting`() {
    author.attemptsTo(UploadNewPost.containing("{\"title\":\"foo\",\"body\":\"bar\",\"userId\":1}"))
    author.should(seeThatResponse {
      it.statusCode(HttpStatus.SC_CREATED).body("id", CoreMatchers.`is`(101))
    })
  }

  companion object {
    private lateinit var author: Actor

    @JvmStatic
    @BeforeClass
    fun setUp() {
      author = Actor("author")
      author.can(CallAnApi.at("https://jsonplaceholder.typicode.com"))
    }
  }
}